### PR TITLE
chore(ci): Add capability to build and push test images to GCR.

### PR DESCRIPTION
**Pull request by b4handjr**
Adds the steps needed to build and push our containerized tests to GCR.

[SYNC-4559](https://mozilla-hub.atlassian.net/browse/SYNC-4559)

[SYNC-4559]: https://mozilla-hub.atlassian.net/browse/SYNC-559?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .circleci/config.yml
Comment (emaydeck-mozilla) [inline .circleci/config.yml:343]: This probably isn't necessary since we're not specifying a default for either of the build jobs and we're passing this parameter from every job that calls it.

**Review by emaydeck-mozilla — APPROVED** [review]
This looks good to me. If you want, you can test this out by having it temporarily filter against your branch instead of `master` in the job definitions. Once your push up a commit in your branch, CircleCI should try to build and deploy the images and I can confirm whether or not they were successfully pushed to GAR.

**Review by jrconlin — APPROVED** [review]

File: .circleci/config.yml
Comment (Trinaa) [inline .circleci/config.yml:483]: This is a departure from the way we have the PR builds setup currently. We usually build the load-test image in PR builds without deploying, so developers can be signaled before they merge if they have introduced a breaking change. I suggest to remove this filter and perhaps the one for the notification test container as well for the same reason.

**Review by Trinaa — DISMISSED** [review]
Pre-approving so that you are not blocked over the holidays @b4handjr.

I suggest to take this opportunity to rename the 'notification' tests, which is not a commonly recognized test category or level , to 'end-to-end' tests.

**Pull request merged**
